supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/labels/Label.scala

```diff
@@ -0,0 +1,59 @@
+package cromwell.backend.impl.jes.labels
+
+/**
+  * Construct via Label.safeLabel
+  */
+case class Label private[labels](key: String, value: String) {
+
+  def +(that: Label): Labels = Labels(Vector(this, that))
+}
+
+object Label {
+
+  def validate(s: String) = "[a-z]([-a-z0-9]*[a-z0-9])?".r.pattern.matcher(s).matches && s.length <= 63
+
+  /**
+    * Enforce the constraint:
+    *  - Must fix the regex [a-z][-a-z0-9]*[a-z0-9]
+    *  - Must be shorter than 63 characters total
+    */
+  def safeName(mainText: String) = {
+
+    if (validate(mainText)) {
+      mainText
+    } else {
+      def appendSafe(current: String, nextChar: Char): String = {
+        nextChar match {
+          case c if c.isLetterOrDigit || c == '-' => current + c.toLower
+          case _ => current + '-'
+        }
+      }
+
+      val foldResult = mainText.toCharArray.foldLeft("")(appendSafe)
+
+      val startsValid = foldResult.headOption.exists(_.isLetter)
+      val finalChar = if (foldResult.length == 0) {
+        None
+      } else {
+        Some(foldResult.charAt(foldResult.length - 1))
+      }
+      val endsValid = finalChar.exists(_.isLetterOrDigit)
+
+      val validStart = if (startsValid) foldResult else "x--" + foldResult
+      val validStartAndEnd = if (endsValid) validStart else validStart + "--x"
+
+      val length = validStartAndEnd.length
+      val tooLong = length > 63
+
+      if (tooLong) {
+        validStartAndEnd.substring(0, 30) + "---" + validStartAndEnd.substring(length - 30, length)
```

Comment:
Pinging @kcibul and/or @dvoet to give a thumbs up to the scheme (see [here](https://github.com/broadinstitute/cromwell/tree/cjl_workflow_id_label#label-escaping-and-padding))